docs/performance/storage.md

```diff
@@ -0,0 +1,316 @@
+it will have no impact on performance and might make sense to reduce its storage footprint.
+
+Things that can be done:
+
+- disable indexing
+- disable the columnar store
+- change the compression algorithm
+- review the data schema
+
+> Disk size improvements can vary depending on the data types, schema and even disk manufacturer. This example is intended for
+> illustrative purposes only.
+
+## Disable indexing
+
+By default, CrateDB creates indexes on every column; this can be disabled when creating the table:
+
+```sql
+CREATE TABLE taxi
+(
+    "VendorID"              BIGINT INDEX OFF,
+    "tpep_pickup_datetime"  TIMESTAMP WITHOUT TIME ZONE INDEX OFF,
+    "tpep_dropoff_datetime" TIMESTAMP WITHOUT TIME ZONE INDEX OFF,
+    ...
+)
+```
+
+The index can only be disabled when the table is created, if the table already exists and it cannot
+be deleted it will have to be re-created.
+
+One of the ways of re-creating a table is by `renaming`, for example:
+
+1. Rename table `taxi` (with INDEX) to `taxi_deleteme` with:
+
+```sql
+ALTER TABLE "taxi"
+    RENAME TO "taxi_deleteme"
+```
+
+2. Create the new table named `taxi` with indexes off.
+
+3. Copy data from `taxi_deleteme` to `taxi`.
+
+```sql
+INSERT INTO "taxi" (SELECT * FROM "taxi_deleteme")
+```
+
+4. Delete `taxi_deleteme` with:
+
+```sql
+DROP TABLE "taxi_deleteme"
+```
+
+> WARNING: Dropping the table deletes the data, make sure that the copy was done correctly.
+
+> INFO: Indexes cannot be re-added after table creation.
+
+### Effects on storage
+
+| avg_mb_per_shard | total_mb |
+|------------------|----------|
+| 53               | 635      |
+
+Data was reduced `18%`
+
+## Disable the columnar store.
+
+The columnar store can be disabled at table creation with:
+
+```sql
+CREATE TABLE IF NOT EXISTS "doc"."taxi_nocolumnstore"(
+    "VendorID" BIGINT STORAGE WITH(
+        columnstore = false
+   ) ,
+   "tpep_pickup_datetime" TIMESTAMP WITHOUT TIME ZONE STORAGE WITH (
+      columnstore = false
+   ),
+   "tpep_dropoff_datetime" TIMESTAMP WITHOUT TIME ZONE STORAGE WITH (
+      columnstore = false
+   ),
+    ...
+)
+```
+
+> As with indexing, it can only be turned off at table creation time.
+
+### Effects on storage
+
+| avg_mb_per_shard | total_mb |
+|------------------|----------|
+| 53               | 639      |
+
+Data was reduced: `18%`, similar to `no_index`.
+
+## Changing the compression algorithm
+
+Data is compressed when it is stored on disk, two options are available `default` (LZ4) and
+`best_compression`.
+
+`best_compression` might be less performant on certain queries, but it has less storage footprint.
+
+You can change it via table definition:
+
+```sql
+CREATE TABLE IF NOT EXISTS "doc"."taxi_january_nocolumnstore" (
+    "VendorID" BIGINT,
+    ...
+) WITH (codec = 'best_compression')
+```
+
+### Effects on storage
+
+| avg_mb_per_shard | total_mb |
+|------------------|----------|
+| 44               | 536      |
+
+Data was reduced: `31.25%`
```

Comment:
Above, you report with two significant digits and here with four. I suggest that you use three so the `18%` is more nuanced.